tools/query_impc/impc_tool.xml

```diff
@@ -1,313 +1,311 @@
+                <option value="8">8 - Extract all genes names and ID measured in a specific IMPReSS pipeline</option>
+                <option value="9">9 - Extract all genes and corresponding phenotypes related to a particular top level phenotype category</option>
+            </param>
+            <when value="1">
+                <conditional name="inp_q">
+                    <expand macro="inputType" />
+                    <when value="str">
+                        <param name="input" type="text" label="Input gene" help="Enter a single MGI gene ID or gene symbol"/>
+                    </when>
+                    <when value="txt">
+                        <param name="input" type="data" format="tabular,txt" label="Input file" help="Enter a txt file with the Gene MGI id or gene symbol"/>
+                        <expand macro="selectSeparator" />
+                    </when>
+                </conditional>
+                <param name="head" type="boolean" checked="true" truevalue="True" falsevalue="False" label="Choose if include the header in the output. The default value is True."/>
+            </when>
+            <when value="2">
+                <conditional name="inp_q">
+                    <expand macro="inputType" />
+                    <when value="str">
+                        <param name="input" type="text" label="Input phenotype or set of phenotypes" help="Enter a single MP/HP term id or a list dividing each id with a comma (without spaces)"/>
+                    </when>
+                    <when value="txt">
+                        <param name="input" type="data" format="data,tabular,txt" label="Input file" help="Enter a txt file with the MP/HP terms"/>
+                        <expand macro="selectSeparator" />
+                    </when>
+                </conditional>
+                <param name="head" type="boolean" checked="true" truevalue="True" falsevalue="False" label="Choose if include the header in the output. The default value is True."/>
+                <expand macro="outputType" />
+            </when>
+            <when value="3">
+                <conditional name="inp_q">
+                    <expand macro="inputType" />
+                    <when value="str">
+                        <param name="input" type="text" label="Input gene or set of genes" help="Enter a single MGI gene id (or gene symbol) or a list dividing each id with a comma (without spaces)"/>
+                    </when>
+                    <when value="txt">
+                        <param name="input" type="data" format="data,tabular,txt" label="Input file" help="Enter a txt file with the genes MGI ids or symbols"/>
+                        <expand macro="selectSeparator" />
+                    </when>
+                </conditional>
+                <param name="head" type="boolean" checked="true" truevalue="True" falsevalue="False" label="Choose if include the header in the output. The default value is True."/>
```

Comment:
This boolean could be a macro as well. The label also could be improved. And some parts could be moved into help=""